README.md
q3aServ - Create a Quake 3 server easily on Linux
=================================================

### Dependences:

  * Quake 3 Engine -> [https://ioquake3.org/](https://ioquake3.org/)
  * OSP Mod `(Included)` -> [http://www.moddb.com/mods/osp/downloads/osp-mod-103a-full](http://www.moddb.com/mods/osp/downloads/osp-mod-103a-full)
  * CPMA Mod `(Included)` -> [https://www.playmorepromode.org/](https://www.playmorepromode.org/)
  * Quake 3 Base -> [http://store.steampowered.com/app/2200/Quake_III_Arena/](http://store.steampowered.com/app/2200/Quake_III_Arena/)
  * Extra maps `(Included)` -> [https://www.lvlworld.com/](https://www.lvlworld.com/)

## Previous steps:

  * Copy all files and folders in root folder of the game.
  * Edit the file `q3aServ.sh`.
  * Change the following variables to the appropriate ones in your case:
  
 ```shell
 RUN_PATH="/opt/Games/quake3"
 RUN_BINARY="ioq3ded.x86_64"
 STATUS_PATH="/var/www/html/quake3/status.txt"
 ```

- Finally, execute the script with the appropriate parameters:
  
 ```cpp
 $ ./q3aServ.sh <option> [1-5]
 ```
    
### Sintax (Basic):

 ```shell
 $ ./q3aServ.sh
  
 Quake 3 Server Script v4.3
  
 Config.files: /opt/Games/quake3/server/*.cfg
 Log.files: /home/q3aql/.q3a/baseq3/*.log
 Status: /var/www/html/quake3/status.txt
  
 Syntax: q3aServ.sh <option> [1-5]
  
 Available options:
  
 CTF - Start Capture The Flag
 TD  - Start Team Deathmatch
 FFA - Start Free For All
 TOU - Start Tournament
 CG  - Custom Game
 CL  - Clear Logs
 ST  - Stop Current Server
 ```
 
* An example command would be the following:

 ```shell
 $ ./q3aServ.sh FFA 3
 ```
 
* The number `3` indicates the map selection group. Each number will automatically choose certain maps.
    
* Default settings of basic mode are the following:

 ```shell
 BOTS_LEVEL="2"
 MIN_PLAYERS="5"
 TIME_LIMIT="25"
 FLAGS_LIMIT="10"
 FRAGS_LIMIT="150"
 TOU_FRAGS_LIMIT="50"
 ```
    
_Note: You can change this settings editing the file `q3aServ.sh` or using the custom game `CG`._
    
## Sintax (Custom Game):

* With the option `CG` the syntax of the command is extended to be able to customize more game options:

 ```shell
 $ ./q3aServ.sh CG

 Syntax: q3aServ.sh CG [map-selection] [gametype] [minplayers] [bot-level] [time-limit] [flags/frags-limit]

 Available options:

 * map-selection -> 1-5
 * gametype -> CTF/FFA/TD/TOU/FT/CTFI/TDI/FFAI/CA/CAI
 * minplayers -> 0-10
 * bot-level -> 1-5
 * time-limit -> 0-999
 * flags/frags-limit -> 0-999

 Abbreviations:

 - CTF  = Capture The Flag
 - FFA  = Free For All
 - TD   = Team Deathmatch
 - TOU  = Tournament
 - FT   = Freeze Tag TD (OSP)
 - CFTI = Capture The Flag Instagib (OSP)
 - TDI  = Team Deatchmatch Instagib (OSP)
 - FFAI = Free For All Instagib (OSP)
 - CA   = Clan Arena (OSP)
 - CAI  = Clan Arena Instagib (OSP)
 ```
 
* An example command would be the following:

 ```shell
 $ ./q3aServ.sh CG 2 TDI 10 2 30 100
 ```


cpma/changelog.txt

This changelog is the canonical source of information for CPMA.
Any document that disagrees with it is almost certainly wrong.

See the end of this file for known issues.


Notes for version 1.48 (27 Jul 10)

add: r_unreferee/unreferee referee's command

add: ch_gauntlet 1/0 in hud(WeaponList)

add: cg_crosshairSize can set the width and height of the crosshair (format "WxH").
    For example, cg_crosshairSize '32x48'

add: cg_simpleItems can vary from 0.1 to 2.0 as a multiplier size of an item icons.
	For exmaple, cg_simpleItems 1.5 will increase the size of icons to 50%

add: cg_noAmmoChange 2 does not switch to another weapon if no ammo

add: invalid characters in cg_altlightning value disable LG beam

add: cq3/dev crouchsliding

add: color code ^8(LtGrey) and ^9(Violet), except console messages

chg: votesystem
	always ignore speconly
	in 1V1 mode ignore specs except remove or kick votables
	in 1V1 mode do not ignore specs for map, gameplay, match, mode or restart votables during non-warmup
	in 1V1 mode make a decision by timer, if vote counts are not equal and more than two
	in TEAM-based modes ignore specs while teams have players
	do not ignore specs for referee votable, only spec can be elected referee

chg: referee drops his status when joining the match

chg: remove vote windelay, vote_allow_windelay

chg: 'players' command can show packetlost

chg: 'players' command for rcon always show ready/notready status

chg: values mode_idletime may change only in the range of 1 to 60

chg: increase sv_fps maximum value to 40 for cnq3

chg: ban the 'kill' command to CTFS

chg: cq3 RL speed 950, splash radius 110

chg: DEV use PM2 RG (remove scaling damage)

chg: cq3 RG fixed damage 80 and scaling knockback from 125% to 75%. fastrail on.

chg: cq3 items height is 50

chg: cq3 tweak doublejumps

chg: player stats resets to the beginning of the match for CA/CTFS/FTAG

chg: fix roundwarmup

chg: CTFS can use roundwarmup

chg: remove warnings for empty s_announcer

fix: a few bot ai bugs

fix: cg_altlightning enemy color equal mates color

fix: armorbar, ammobar, healthbar cannot display big value (400 health or 400 armor)

fix: don't count suicide in warmup

fix: commands referee, including the options, available in the server console

fix: PM_CmdScale '-128/127 bug'

fix: ancient callvote security exploit (id bug)
	i thought we fixed these years ago, but one case snuck through  :(


Notes for version 1.47 (22 Apr 09)

add: roundwarmup for CA/DA

add: negative cg_trueLightning is no beam

add: cq3 new hitsounds

add: voulcaine's maps layouts: pro-q3dm6cq3, q3tourney2cq3, pro-q3tourney4cq3,
     nodm9acq3, hub3aeroq3cq3, ztn3tourney1cq3, q3dm7cq3

add: LG impact texture (thanks voulcaine)

add: gp as an alias for gameplay for callvote command

add: wr as an alias for weaponrespawn for callvote command

add: duration match to xml stats

chg: 'tell' without player number agrument sent message to all spectators

chg: remove tell_target and tell_attacker

chg: #N use client netname when nick is empty

chg: Coaches are able to use the SuperHUD (for MVD)

chg: Ignore spec votes in teamplay modes (Teams must have players).

chg: The chat macro #I ignore own flag

chg: Silent ref kicks of specs

chg: fastrail back

chg: increase max players in player_file/filter_file to 100

chg: use KISS32 random generator

chg: cg_altlightning is string('123' or '12' or '1'). First character is player lightning,
     Second character is enemy lightning. Third character is mates lightning. 
     First character is default for enemy and mates lightning if not specified.

chg: cg_altlightning 3 with 1.44 render system

chg: cg_drawgun 0 is static(no bobbing)

chg: health and armour votes allow up to 400

chg: DA defaults to 250/250 spawns

chg: a few tweak cq3 ground friction and acceleration

chg: add micro-dj for cq3

chg: (1v1, 2v2, ctf, ftag) cq3 player respawn with ammo. 2 SSG, 2 RL, 20 LG, 3 GL, 3 RG

chg: cq3 items height is 46

chg: cq3 dead player drop ammo like CPM

chg: cq3 respawns are biased away from corpse (except nearest spawnpoint)

chg: cq3 gauntlet knockback is 0.5

chg: cq3 MG - ammo 60/10/80; reload time 90ms; increase spread

chg: cq3 SSG - new pattern; ammo 8/5/25

chg: cq3 PG ammo 40/20/80

chg: cq3 GL use PM2 settings, ammo 7/5/25

chg: cq3 RL speed 1000 ups, splash radius 100, knockback  110%, ammo 8/5/25

chg: cq3 LG scaling knockback from 125% to 75%. ammo 60/40/140

chg: cq3 RG damage (100,90,80); scaling knockback (100%-50%). ammo 7/5/25

chg: cq3 BFG ammo 10/10/25; damage 125; knockback 150%; BFG shots are "fat"

chg: cq3 increase knockback for rj/gj and plasmaclimbing

chg: cq3 different protection (0.5/0.66/0.75) and different decay (0.5/1/1.5) for GA/YA/RA

fix: wrong ammo and flag weight after cv gameplay or cv mode(mode file with gameplay) 

fix: gameplay initialization refresh current game mode 

fix: file handle leaks

fix: CA/FTAG startrespawn 1 

fix: "humming" weapons could pan and crackle on stairs sometimes

fix: flagrun times weren't properly adjusted for pauses

fix: stupidly long text decorations could crash the hud parser


Notes for version 1.46 (26 Apr 08)

chg: mode_current tagged as serverinfo for MAXBrowse / ASE / etc

chg: FFA mode uses 5s weapon respawns

chg: reverted to 1.43 cg_altLightning

chg: removed hemostick's RL explosions
	they're derived from something other than Q3 apparently

chg: removed uriel's wings - they're annoyingly defective

fix: listen servers hanging sometimes on map load

fix: loud LG impact sounds with correctly-functioning Q3 clients


Notes for version 1.45 (18 Feb 08)

add: callvote remove is back

chg: RG low ammo warning

chg: CQ3 pg/sg ammobox

chg: CQ3 RG scaling from 100 to 80

chg: CQ3 hitsound from CPM


q3aServ.sh
#!/bin/bash

# --------------------------------------------
# Quake 3 Server Script                      |
# Created by q3aql (q3aql@protonmail.ch)     |
# Licensed by GPL v2.0                       |
# Last update: 13-03-2021                    |
# --------------------------------------------
VERSION="4.3"

# Path with binary and game files
RUN_PATH="/opt/Games/quake3"
cd ${RUN_PATH}

# Binary of server
RUN_BINARY="ioq3ded.x86_64"

# Baseq3 folder path
BASEQ3_PATH="${HOME}/.q3a/baseq3"
mkdir -p ${HOME}/.q3a/baseq3

# Basic configs
BOTS_LEVEL="2"
MIN_PLAYERS="5"
TIME_LIMIT="25"
FLAGS_LIMIT="10"
FRAGS_LIMIT="150"
TOU_FRAGS_LIMIT="50"
RCON_PASSWORD='"secret"'

# Status of server
STATUS_PATH="/var/www/html/quake3/status.txt"
# Note: Change to '/dev/null' if you want disable it.

# Map selection
expr ${2} + 1 &> /dev/null
mapSelection=$?
if [ -z "${2}" ] ; then
	MAP_SELECT=$(expr ${RANDOM} % 5 + 1)
else
	if [ ${mapSelection} -ne 0 ] ; then
		echo ""
		echo "<<[Error: Map-selection only must be numbers between 1 and 5]>>"
		echo ""
		exit
	elif [ ${2} -lt 1 ] || [ ${2} -gt 5 ] ; then
		echo ""
		echo "<<[Error: Map-selection must be between 1 and 5]>>"
		echo ""
		exit
	else
		MAP_SELECT=${2}
	fi
fi

# Function to stop a previous server and
# remove the config. files
function clear_server() {
	killall -9 ${RUN_BINARY} 2> /dev/null
	rm -rf ${BASEQ3_PATH}/basic.cfg
	rm -rf ${BASEQ3_PATH}/server*.cfg
	rm -rf ${BASEQ3_PATH}/bots*.cfg
	rm -rf ${BASEQ3_PATH}/levels*.cfg
}

# Function to remove the server logs.
function clear_server_logs() {
	echo > ${BASEQ3_PATH}/qconsole.log
	echo > ${BASEQ3_PATH}/server.log
}

# Function to show correct syntax in
# custom game.
function custom_game_menu() {
	echo ""
	echo "Syntax: q3aServ.sh CG [map-selection] [gametype] [minplayers] [bot-level] [time-limit] [flags/frags-limit]"
	echo ""
	echo "Available options:"
	echo ""
	echo "* map-selection -> 1-5"
	echo "* gametype -> CTF/FFA/TD/TOU/FT/CTFI/TDI/FFAI/CA/CAI"
	echo "* minplayers -> 0-10"
	echo "* bot-level -> 1-5"
	echo "* time-limit -> 0-999"
	echo "* flags/frags-limit -> 0-999"
	echo ""
	echo "Abbreviations:"
	echo ""
	echo "- CTF  = Capture The Flag"
	echo "- FFA  = Free For All"
	echo "- TD   = Team Deathmatch"
	echo "- TOU  = Tournament"
	echo "- FT   = Freeze Tag TD (OSP)"
	echo "- CFTI = Capture The Flag Instagib (OSP)"
	echo "- TDI  = Team Deatchmatch Instagib (OSP)"
	echo "- FFAI = Free For All Instagib (OSP)"
	echo "- CA   = Clan Arena (OSP)"
	echo "- CAI  = Clan Arena Instagib (OSP)"
	echo ""
}

# Function to check correct gametypes.
function check_gametype() {
	if [ "${1}" == "CTF" ] ; then
		echo "Gametype selected = CTF"
	elif [ "${1}" == "FFA" ] ; then
		echo "Gametype selected = FFA"
	elif [ "${1}" == "TD" ] ; then
		echo "Gametype selected = TD"
	elif [ "${1}" == "TOU" ] ; then
		echo "Gametype selected = TOU"
	elif [ "${1}" == "FT" ] ; then
		echo "Gametype selected = FT"
	elif [ "${1}" == "CTFI" ] ; then
		echo "Gametype selected = CTFI"
	elif [ "${1}" == "TDI" ] ; then
		echo "Gametype selected = TDI"
	elif [ "${1}" == "FFAI" ] ; then
		echo "Gametype selected = FFAI"
	elif [ "${1}" == "CA" ] ; then
		echo "Gametype selected = CA"
	elif [ "${1}" == "CAI" ] ; then
		echo "Gametype selected = CAI"
	elif [ -z "${1}" ] ; then
		echo "Empty" > /dev/null
	else
		echo ""
		echo "<<[Error: Gametype invalid]>>"
		custom_game_menu
		exit
	fi
}

# Available options
case ${1} in
	CTF|ctf)
	# Initialize "Capture The Flag" session
	clear_server
	cp server/basic.cfg ${BASEQ3_PATH}
	cp server/serverCTF.cfg ${BASEQ3_PATH}
	cp server/bots.cfg ${BASEQ3_PATH}
	cp server/levels/levelsCTF-${MAP_SELECT}.cfg ${BASEQ3_PATH}
	echo "seta g_spskill ${BOTS_LEVEL}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta bot_minplayers ${MIN_PLAYERS}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta timelimit ${TIME_LIMIT}" >> ${BASEQ3_PATH}/serverCTF.cfg
	echo "seta capturelimit ${FLAGS_LIMIT}" >> ${BASEQ3_PATH}/serverCTF.cfg
	echo "seta rconpassword ${RCON_PASSWORD}" >> ${BASEQ3_PATH}/serverCTF.cfg
	echo "Running Capture The Flag (${MAP_SELECT})" > ${STATUS_PATH}
	./${RUN_BINARY} +exec basic.cfg +exec serverCTF.cfg +exec levelsCTF-${MAP_SELECT}.cfg +exec bots.cfg
	#clear_server
	echo "Stopped" > ${STATUS_PATH}
	;;
	TD|td)
	# Initialize "Team Deathmatch" session
	clear_server
	cp server/basic.cfg ${BASEQ3_PATH}
	cp server/serverTD.cfg ${BASEQ3_PATH}
	cp server/bots.cfg ${BASEQ3_PATH}
	cp server/levels/levelsTD-${MAP_SELECT}.cfg ${BASEQ3_PATH}
	echo "seta g_spskill ${BOTS_LEVEL}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta bot_minplayers ${MIN_PLAYERS}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta timelimit ${TIME_LIMIT}" >> ${BASEQ3_PATH}/serverTD.cfg
	echo "seta fraglimit ${FRAGS_LIMIT}" >> ${BASEQ3_PATH}/serverTD.cfg
	echo "seta rconpassword ${RCON_PASSWORD}" >> ${BASEQ3_PATH}/serverTD.cfg
	echo "Running Team Deathmatch (${MAP_SELECT})" > ${STATUS_PATH}
	./${RUN_BINARY} +exec basic.cfg +exec serverTD.cfg +exec levelsTD-${MAP_SELECT}.cfg +exec bots.cfg
	#clear_server
	echo "Stopped" > ${STATUS_PATH}
	;;
	FFA|ffa)
	# Initialize "Free For All" session
	clear_server
	cp server/basic.cfg ${BASEQ3_PATH}
	cp server/serverFFA.cfg ${BASEQ3_PATH}
	cp server/bots.cfg ${BASEQ3_PATH}
	cp server/levels/levelsFFA-${MAP_SELECT}.cfg ${BASEQ3_PATH}
	echo "seta g_spskill ${BOTS_LEVEL}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta bot_minplayers ${MIN_PLAYERS}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta timelimit ${TIME_LIMIT}" >> ${BASEQ3_PATH}/serverFFA.cfg
	echo "seta fraglimit ${FRAGS_LIMIT}" >> ${BASEQ3_PATH}/serverFFA.cfg
	echo "seta rconpassword ${RCON_PASSWORD}" >> ${BASEQ3_PATH}/serverFFA.cfg
	echo "Running Free For All (${MAP_SELECT})" > ${STATUS_PATH}
	./${RUN_BINARY} +exec basic.cfg +exec serverFFA.cfg +exec levelsFFA-${MAP_SELECT}.cfg +exec bots.cfg
	#clear_server
	echo "Stopped" > ${STATUS_PATH}
	;;
	TOU|tou|tourney)
	# Initialize "Tournament" session
	clear_server
	cp server/basic.cfg ${BASEQ3_PATH}
	cp server/serverTOU.cfg ${BASEQ3_PATH}
	cp server/bots.cfg ${BASEQ3_PATH}
	cp server/levels/levelsTOU-${MAP_SELECT}.cfg ${BASEQ3_PATH}
	echo "seta g_spskill ${BOTS_LEVEL}" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta bot_minplayers 2" >> ${BASEQ3_PATH}/bots.cfg
	echo "seta timelimit ${TIME_LIMIT}" >> ${BASEQ3_PATH}/serverTOU.cfg
	echo "seta fraglimit ${TOU_FRAGS_LIMIT}" >> ${BASEQ3_PATH}/serverTOU.cfg
	echo "seta rconpassword ${RCON_PASSWORD}" >> ${BASEQ3_PATH}/serverTOU.cfg
	echo "Running Tournament (${MAP_SELECT})" > ${STATUS_PATH}
	./${RUN_BINARY} +exec basic.cfg +exec serverTOU.cfg +exec levelsTOU-${MAP_SELECT}.cfg +exec bots.cfg
	#clear_server
	echo "Stopped" > ${STATUS_PATH}
	;;
	CG|cg|custom)
	# Initialize "Custom Game" session
	check_gametype ${3}
	expr ${4} + 1 &> /dev/null
	minPlayers=$?
	expr ${5} + 1 &> /dev/null
	botLevel=$?
	expr ${6} + 1 &> /dev/null
	timeLimit=$?
	expr ${7} + 1 &> /dev/null
	FFLimit=$?
	if [ -z "${2}" ] ; then
		custom_game_menu
	elif [ "${2}" == "-h" ] || [ "${2}" == "--help" ] ; then
		custom_game_menu
	elif [ -z "${4}" ] ; then
		echo ""
		echo "<<[Error: Minplayers must be specified]>>"
		custom_game_menu
	elif [ ${minPlayers} -ne 0 ] ; then
		echo ""
		echo "<<[Error: Minplayers only must be numbers between 0 and 10]>>"
		custom_game_menu
	elif [ ${4} -lt 0 ] || [ ${4} -gt 10 ] ; then
		echo ""
		echo "<<[Error: Minplayers must be between 0 and 10]>>"
		custom_game_menu
	elif [ -z "${5}" ] ; then
		echo ""
		echo "<<[Error: Bot-level must be specified]>>"
		custom_game_menu
	elif [ ${botLevel} -ne 0 ] ; then
		echo ""
		echo "<<[Error: Bot-level only must be numbers between 1 and 5]>>"
		custom_game_menu
	elif [ ${5} -lt 1 ] || [ ${5} -gt 5 ] ; then
		echo ""
		echo "<<[Error: Bot-level must be between 1 and 5]>>"
		custom_game_menu
	elif [ -z "${6}" ] ; then
		echo ""
		echo "<<[Error: Time-limit must be specified]>>"
		custom_game_menu
	elif [ ${timeLimit} -ne 0 ] ; then
		echo ""
		echo "<<[Error: Time-limit only must be numbers between 0 and 999]>>"
		custom_game_menu
	elif [ ${6} -lt 0 ] || [ ${6} -gt 999 ] ; then
		echo ""
		echo "<<[Error: Time-limit must be between 0 and 999]>>"
		custom_game_menu
	elif [ -z "${7}" ] ; then
		echo ""
		echo "<<[Error: Flags/frags-limit must be specified]>>"
		custom_game_menu
	elif [ ${FFLimit} -ne 0 ] ; then
		echo ""
		echo "<<[Error: Flags/frags-limit only must be numbers between 0 and 999]>>"
		custom_game_menu
	elif [ ${7} -lt 0 ] || [ ${7} -gt 999 ] ; then
		echo ""
		echo "<<[Error: Flags/frags-limit must be between 0 and 999]>>"
		custom_game_menu
	else
		clear_server
		cp server/basic.cfg ${BASEQ3_PATH}
		cp server/server${3}.cfg ${BASEQ3_PATH}
		cp server/bots.cfg ${BASEQ3_PATH}
		cp server/levels/levels${3}-${2}.cfg ${BASEQ3_PATH}
		echo "seta g_spskill ${5}" >> ${BASEQ3_PATH}/bots.cfg
		echo "seta bot_minplayers ${4}" >> ${BASEQ3_PATH}/bots.cfg
		echo "seta timelimit ${6}" >> ${BASEQ3_PATH}/server${3}.cfg
		if [ "${3}" == "CTF" ] ; then
			echo "seta capturelimit ${7}" >> ${BASEQ3_PATH}/server${3}.cfg
		elif [ "${3}" == "CTFI" ] ; then
			echo "seta capturelimit ${7}" >> ${BASEQ3_PATH}/server${3}.cfg
		else
			echo "seta fraglimit ${7}" >> ${BASEQ3_PATH}/server${3}.cfg
		fi
		echo "seta rconpassword ${RCON_PASSWORD}" >> ${BASEQ3_PATH}/server${3}.cfg
		echo "Running ${3} (Custom Game) ($2)" > ${STATUS_PATH}
		if [ "${3}" == "FT" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		elif [ "${3}" == "CTFI" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		elif [ "${3}" == "TDI" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		elif [ "${3}" == "FFAI" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		elif [ "${3}" == "CA" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		elif [ "${3}" == "CAI" ] ; then
			./${RUN_BINARY} +set fs_game osp +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		else
			./${RUN_BINARY} +exec basic.cfg +exec server${3}.cfg +exec levels${3}-${2}.cfg +exec bots.cfg
		fi
		#clear_server
		echo "Stopped" > ${STATUS_PATH}
	fi
	;;
	ST|st|stop|STOP)
	# Close previous sessions
	clear_server
	echo "Stopped" > ${STATUS_PATH}
	;;
	CL|cl)
	# Clear all server logs
	clear_server_logs
	;;
	*)
	# Show menu
	echo ""
	echo "Quake 3 Server Script v${VERSION}"
	echo ""
	echo "Config.files: ${RUN_PATH}/server/*.cfg"
	echo "Log.files: ${BASEQ3_PATH}/*.log"
	echo "Status: ${STATUS_PATH}"
	echo ""
	echo "Syntax: q3aServ.sh <option> [1-5]"
	echo ""
	echo "Available options:"
	echo ""
	echo "CTF - Start Capture The Flag"
	echo "TD  - Start Team Deathmatch"
	echo "FFA - Start Free For All"
	echo "TOU - Start Tournament"
	echo "CG  - Custom Game"
	echo "CL  - Clear Logs"
	echo "ST  - Stop Current Server"
	echo ""
esac
